parsec/core/logged_core.py

```diff
@@ -41,6 +42,9 @@ class LoggedCore:
 async def logged_core_factory(
     config: CoreConfig, device: LocalDevice, event_bus: Optional[EventBus] = None
 ):
+    # Add task monitoring instrument
+    trio.hazmat.add_instrument(TaskMonitoringInstrument())
```

Comment:
Should be moved elsewhere given the instrument is not really related to this function (and we are going to push multiple instruments if we call this function multiple time, like when the user does login/logout in the gui)

We should have instead:
- `@fixture(autouse=True)` defined in `tests/conftest.py` that returns the instrument (pytest-trio automatically collect instruments returned by fixtures and pass them to the `trio.run()`)
- for the cli, create a new `parsec.utils.trio_run` function that should be use instead of the original `trio.run` and `trio_asyncio.run` (so our `trio_run` should have a `with_asyncio=False` param to determine if we need to run a trio_asyncio loop on top of the trio loop)